src/precice/tests/SerialTests.cpp

```diff
@@ -1050,10 +1107,14 @@ BOOST_AUTO_TEST_CASE(AccessReceivedMeshImplicit)
       }
 
       // Expected data according to the writeData
-      std::vector<double> expectedData({10, 11, 12, 13});
-      BOOST_TEST(expectedData == readData);
+      std::vector<double> expectedData{10, 11, 12, 13};
+      BOOST_TEST(expectedData[0] == readData[0]);
+      BOOST_TEST(expectedData[1] == readData[1]);
+      BOOST_TEST(expectedData[2] == readData[2]);
+      BOOST_TEST(expectedData[3] == readData[3]);
+      BOOST_TEST(expectedData.size() == readData.size());
+      BOOST_TEST(expectedData.size() == 4);
```

Comment:
I'm unsure why this change is required. `BOOST_TEST(expectedData == readData)` should work, but it does not.